sympy/polys/polyroots.py

```diff
@@ -1094,6 +1141,15 @@ def _try_heuristics(f):
     # adding zero roots after non-trivial roots have been translated
     result.update(zeros)
 
+    if strict and sum(result.values()) < f.degree():
+        raise UnsolvableFactorError(filldedent('''
+            Strict mode: some solution components cannot be
+            expressed in radicals, so a complete list of solutions
```

Comment:
It doesn't make sense to say "solution components" here when we're dealing with univiariate polynomials. Rather "some factors cannot be solved in radicals".